docs/guide/input-file-upload.md

```diff
@@ -55,6 +55,12 @@ the [Core Validators](tutorial-core-validators.md#file) section for more details
   implemented via [[yii\validators\ImageValidator]] which verifies if an attribute has received a valid image 
   that can be then either saved or processed using the [Imagine Extension](https://github.com/yiisoft/yii2-imagine).
 
+If you need to validate your model after use of `saveAs` method, especially with the `image` validator, you need to add `false` paramether to `saveAs`, because this method deletes an uploaded file during saving by default and validator will give out an error.
```

Comment:
If $uploadedfile is a property of ActiveRecord model, every save() method calls validate() method too. Usually I process uploaded file through separate method before saving the db record or in a beforeSave method. I know, that processing forms in AR model isn't the best practice, but it's really often usecase.